### PR TITLE
WebCmdlets:   empty body should not be added for Get Methods

### LINKED ISSUE #15186: Invoke-WebRequest/Invoke-RestMethod: PR #10034 violates RFC7230 and therefore breaks HTTP Desync Mitigation modes
## Background

PR #10034 that fixes issue #9473 and has been introduced in `7.0.0-preview.2`.
This sets a `String.Empty` body for GET requests in `Invoke-WebRequest` and `Invoke-RestMethod` which usually do not have a body.

Unfortunately this also leads .NET to add a `Content-Length: 0` header because of the `String.Empty` HTTP request body.

And this breaks RFC7230, see the official RFC7230 Section 3.3.2: https://tools.ietf.org/html/rfc7230#section-3.3.2 
```
   A user agent SHOULD NOT send a
   Content-Length header field when the request message does not contain
   a payload body and the method semantics do not anticipate such a
   body.
```
Why is this a problem?
* To prevent HTTP Request Smuggling attacks Cloud Service Providers such as AWS and vendors of Application Load Balancers have implemented so-called "HTTP Desync Mitigation modes".
* On AWS the "Strictest Mode" for example blocks all requests that are not RFC7230 conform.
** See https://docs.aws.amazon.com/elasticloadbalancing/latest/application/application-load-balancers.html#desync-mitigation-mode for explanation
* Here is a list of classification reasons that are not RFC7230 conform: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html#classification-reasons
** `GetHeadZeroContentLength` - There is a `Content-Length` header with a value of 0 for a GET or HEAD request.
** This classifies the request as `Acceptable` which is being blocked in "Strictest Mode"
* You cannot use `Invoke-WebRequest` or `Invoke-RestMethod` on AWS hosted websites where this setting has been set
* This will result in a HTTP 400 - Bad Request
* This might also affect other Cloud Service Providers like Azure or Google Cloud Platform, but I cannot confirm that, as I'm only using AWS.

## Steps to reproduce

* On an AWS Application Load Balancer set the HTTP Desync Mitigation Mode to "Strictest Mode"
* Then try a simple `Invoke-WebRequest` on `PowerShell 7.0.0-preview.2` or above:
```powershell
MY C:\> Invoke-WebRequest -Uri "https://mywebsite.example.com/"
Invoke-WebRequest:
400 Bad Request
400 Bad Request
```

## Expected behavior

* Do not set a `Content-Length: 0` header as it is not RFC7230 conform if you do not have a HTTP Request body

## Actual behavior

* PowerShell 7 always sets the HTTP header `Content-Length: 0` for GET/HEAD requests without HTTP Request body and therefore violates RFC7230

## Environment data

* This is reproducible on AWS Lambda with `PowerShell 7.0.0` on Linux as well as on Windows 10 with `PowerShell 7.1.0`

### PR DESCRIPTION
**Pull request by VeVrX**
# PR Summary

Fix #15186 - To comply with RFC7230, empty body should not be added for Get Methods

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 1]
```diff
@@ -1277,8 +1277,10 @@ internal virtual void FillRequestStream(HttpRequestMessage request)
                 }
             }
 
-            // Add the content headers
-            if (request.Content == null)
+            // Add the content headers except for Get Method
```
Comment (TravisEz13) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:1280]: Can you please reference the RFC
```suggestion
            // Add the content headers except for Get Method per RFC7230
```

**Review by TravisEz13 — CHANGES_REQUESTED** [review]
see comments

**Comment by TravisEz13** [comment]
Two tests need to be updated:
https://dev.azure.com/powershell/PowerShell/_build/results?buildId=91101&view=logs&j=6cde3b71-baca-5747-8e0b-9d2766aa716f&t=47203bc9-c6f6-5ea0-f0a9-178d83600668&l=1580

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
Oh, it is dup of #16587 and it seems the fix cover more scenarios but haven't tests. We need resolve the conflict and combine the PRs.

**Comment by SteveL-MSFT** [comment]
Closing this PR as https://github.com/PowerShell/PowerShell/pull/16587 covers this and has necessary test changes

**Comment by iSazonov** [comment]
@VeVrX Thanks for your contribution! Sorry for the PR conflict.

**Pull request closed without merging**